### ISSUE TITLE
LD_PRELOAD confuses cx_freeze ELF dependency parsing

### ISSUE THREAD

**Issue by mutability** [marcelotduarte/cx_Freeze#2064]
TL;DR: If `LD_PRELOAD` is set, cx_freeze will recurse forever trying to resolve ELF dependencies.

Here's an example, invoking cx_freeze within a Debian package-building environment which has `libfakeroot` in `LD_PRELOAD` as a result of the package build happening within `fakeroot`:

```
/build/jenkins/workspace/piaware_dev/pkg-bookworm/debian/venv/bin/python /build/jenkins/workspace/piaware_dev/pkg-bookworm/debian/venv/bin/cxfreeze --target-dir=/build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client /build/jenkins/workspace/piaware_dev/pkg-bookworm/debian/venv/bin/fa-mlat-client
creating directory /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib
copying /usr/lib/aarch64-linux-gnu/libpython3.11.so.1.0 -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libpython3.11.so.1.0
creating directory /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot
copying /usr/lib/aarch64-linux-gnu/libfakeroot/libfakeroot-sysv.so -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot-sysv.so
copying /usr/lib/aarch64-linux-gnu/libeatmydata.so.1.3.0 -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libeatmydata.so.1.3.0
creating directory /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot
copying /usr/lib/aarch64-linux-gnu/libfakeroot/libfakeroot-sysv.so -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot-sysv.so
copying /usr/lib/aarch64-linux-gnu/libeatmydata.so.1.3.0 -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libeatmydata.so.1.3.0
creating directory /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot
copying /usr/lib/aarch64-linux-gnu/libfakeroot/libfakeroot-sysv.so -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot-sysv.so
copying /usr/lib/aarch64-linux-gnu/libeatmydata.so.1.3.0 -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libeatmydata.so.1.3.0
creating directory /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot
copying /usr/lib/aarch64-linux-gnu/libfakeroot/libfakeroot-sysv.so -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot-sysv.so
copying /usr/lib/aarch64-linux-gnu/libeatmydata.so.1.3.0 -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libeatmydata.so.1.3.0
creating directory /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot
copying /usr/lib/aarch64-linux-gnu/libfakeroot/libfakeroot-sysv.so -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot-sysv.so
copying /usr/lib/aarch64-linux-gnu/libeatmydata.so.1.3.0 -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libeatmydata.so.1.3.0
creating directory /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot
copying /usr/lib/aarch64-linux-gnu/libfakeroot/libfakeroot-sysv.so -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot-sysv.so
copying /usr/lib/aarch64-linux-gnu/libeatmydata.so.1.3.0 -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libeatmydata.so.1.3.0
[... several thousand lines later ...]
copying /usr/lib/aarch64-linux-gnu/libfakeroot/libfakeroot-sysv.so -> /build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot-sysv.so
Traceback (most recent call last):
  File "/build/jenkins/workspace/piaware_dev/pkg-bookworm/debian/venv/lib/python3.11/site-packages/cx_Freeze/freezer.py", line 185, in _copy_file
    self._post_copy_hook(source, target, copy_dependent_files)
  File "/build/jenkins/workspace/piaware_dev/pkg-bookworm/debian/venv/lib/python3.11/site-packages/cx_Freeze/freezer.py", line 1219, in _post_copy_hook
    self._copy_file(
  File "/build/jenkins/workspace/piaware_dev/pkg-bookworm/debian/venv/lib/python3.11/site-packages/cx_Freeze/freezer.py", line 185, in _copy_file
    self._post_copy_hook(source, target, copy_dependent_files)
  File "/build/jenkins/workspace/piaware_dev/pkg-bookworm/debian/venv/lib/python3.11/site-packages/cx_Freeze/freezer.py", line 1219, in _post_copy_hook
    self._copy_file(
[...]
  File "/build/jenkins/workspace/piaware_dev/pkg-bookworm/debian/venv/lib/python3.11/site-packages/cx_Freeze/freezer.py", line 1219, in _post_copy_hook
    self._copy_file(
  File "/build/jenkins/workspace/piaware_dev/pkg-bookworm/debian/venv/lib/python3.11/site-packages/cx_Freeze/freezer.py", line 178, in _copy_file
    shutil.copyfile(source, target)
  File "/usr/lib/python3.11/shutil.py", line 258, in copyfile
    with open(dst, 'wb') as fdst:
         ^^^^^^^^^^^^^^^
OSError: [Errno 36] File name too long: '/build/jenkins/workspace/piaware_dev/pkg-bookworm/freeze-mlat-client/lib/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot/libfakeroot-sysv.so'
```

I believe the cause of this is that:

 * cx_freeze invokes `ldd` to follow shared library dependencies
 * `ldd` includes `LD_PRELOAD`-triggered dependencies in its output
 * This is effectively a circular dependency, since libfakeroot then apparently depends on itself (per ldd)

Here's an example of what I mean (on a different system):

```
oliver@travel:~$ echo $LD_PRELOAD

oliver@travel:~$ ldd /bin/bash
	linux-vdso.so.1 (0x00007ffc65db8000)
	libtinfo.so.6 => /lib/x86_64-linux-gnu/libtinfo.so.6 (0x00007fdba4dbf000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fdba4b97000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fdba4f77000)
oliver@travel:~$ fakeroot bash
root@travel:~# echo $LD_PRELOAD
libfakeroot-sysv.so
root@travel:~# ldd /bin/bash
	linux-vdso.so.1 (0x00007fff9b1db000)
	libfakeroot-sysv.so => /usr/lib/x86_64-linux-gnu/libfakeroot/libfakeroot-sysv.so (0x00007f9b47a81000)
	libtinfo.so.6 => /lib/x86_64-linux-gnu/libtinfo.so.6 (0x00007f9b47a2c000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f9b47804000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f9b47bf4000)
root@travel:~# ldd /usr/lib/x86_64-linux-gnu/libfakeroot/libfakeroot-sysv.so
	linux-vdso.so.1 (0x00007fff58bd1000)
	libfakeroot-sysv.so => /usr/lib/x86_64-linux-gnu/libfakeroot/libfakeroot-sysv.so (0x00007f212a554000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f212a309000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f212a576000)
```

cx_freeze should probably exclude `LD_PRELOAD` from dependency-following entirely somehow (perhaps just by clearing that env var before invoking ldd) -- since those dependencies are an environmental thing, not a dependency of the binary itself.

Seen on cx_freeze 6.15.7, Python 3.11.2, Debian 12 (bookworm), arm64 architecture.

**Comment by marcelotduarte** [comment]
[Release 6.15.9](https://github.com/marcelotduarte/cx_Freeze/releases/tag/6.15.9) is out!

Can you test and give feedback?

**Comment by mutability** [comment]
6.15.9 fixes this in my environment. Thanks!
